### DISCUSSION TITLE
How to add description and title via the API

### DISCUSSION THREAD

**Discussion by tacman** [biati-digital/glightbox#516] [Q&A]
How do I add a description (and title, etc.) via the API?  The examples all appear to be data- attributes

```js
        import 'https://cdn.jsdelivr.net/gh/mcstudios/glightbox/dist/js/glightbox.min.js';

        var exampleApi = GLightbox({ selector: null });
        fetch('https://dummyjson.com/products')
            .then(response => response.json())
            .then(data => {
                data.products.forEach(product => {
                    product.images.forEach( i => {
                        exampleApi.insertSlide({
                            href: i,
                            // content: '<p>some html to append in the slide</p>',
                            // description: "A description"
                        });

                    })
                });
                exampleApi.open();

            })
            .catch(err => console.error(err))
 ```

**Comment by tacman** [comment]
```html
    <script type="module">
        import 'https://cdn.jsdelivr.net/gh/mcstudios/glightbox/dist/js/glightbox.min.js';

        var exampleApi = GLightbox({ selector: null });
        exampleApi.insertSlide( {
            'href': 'https://picsum.photos/1200/800',
            'type': 'image',
            'title': 'My Title',
            'description': 'Example',
        });
        exampleApi.open();
</script>
```

throws an error:

stimulus.index-b5b1d00e42695b8959b4a1e94e3bc92a.js:7 application #start
glightbox.min.js:1 Uncaught TypeError: v.getAttribute is not a function
    at e.value (glightbox.min.js:1:52654)
    at glightbox.min.js:1:41042
    at glightbox.min.js:1:22791
    at HTMLImageElement.<anonymous> (glightbox.min.js:1:14623)


glightbox.min.js:1 Uncaught TypeError: v.getAttribute is not a function
    at e.value (glightbox.min.js:1:52654)
    at glightbox.min.js:1:41042
    at glightbox.min.js:1:22791
    at HTMLImageElement.<anonymous> (glightbox.min.js:1:14623)

**Comment by tacman** [comment]
I converted this to a bug report, it's not simply documenting how to do it, but following the instructions in the documentation throws an error.

Thus, more demos!  #513

**Comment by brandjunge** [comment]
I have the same problem and error. I've figured out that it - in my case - only happens in desktop-resolution when I set descPosition to "top" or "bottom". In mobile all four positions are working and the error doesn't prompt in the terminal - but in desktop only "right" and "left" is usable. 

I will work with "right" and change the position within css as a quick workaround...
